### DISCUSSION TITLE
Add easy access to variable member inside a block

### DISCUSSION THREAD

**Discussion by amin67v** [dotnet/csharplang#2854] [Language Ideas]
```
void Foo(Bar bar)
{
    bar.Var1 = bar.Var2 + bar.Var3;
}
```

```
void Foo(Bar bar)
{
    bar.{
        Var1 = Var2 + Var3;
    }
}
```

this is a simple and silly example ofcource, but i think sometimes it saves alot of typing

**Comment by HaloFour** [comment]
Also makes it very difficult to discern what `Var`, `Var2` and `Var3` are.  They could be properties of `bar`, but they could also be other properties in scope.

**Comment by amin67v** [comment]
> Also makes it very difficult to discern what `Var`, `Var2` and `Var3` are. They could be properties of `bar`, but they could also be other properties in scope.

we can get a compiler error like CS0104 in that case, then we have to do this.Var1 or bar.Var1

**Comment by yaakov-h** [comment]
This sounds like the various “with” proposals.

**Comment by GeraudFabien** [comment]
Yes. It sound like VB :)
